contracts/account/extensions/AccountERC7579.sol

```diff
@@ -0,0 +1,214 @@
+        );
+    }
+
+    function _installModule(uint256 moduleTypeId, address module, bytes memory initData) internal virtual {
+        require(supportsModule(moduleTypeId), ERC7579Utils.ERC7579UnsupportedModuleType(moduleTypeId));
+        require(
+            IERC7579Module(module).isModuleType(moduleTypeId),
+            ERC7579Utils.ERC7579MismatchedModuleTypeId(moduleTypeId, module)
+        );
+
+        if (moduleTypeId == MODULE_TYPE_VALIDATOR) {
+            require(_validators.add(module), ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module));
+        } else if (moduleTypeId == MODULE_TYPE_EXECUTOR) {
+            require(_executors.add(module), ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module));
+        } else if (moduleTypeId == MODULE_TYPE_FALLBACK) {
+            bytes4 selector;
+            (selector, initData) = abi.decode(initData, (bytes4, bytes));
+            require(
+                _installFallback(module, selector),
+                ERC7579Utils.ERC7579AlreadyInstalledModule(moduleTypeId, module)
+            );
+        }
+
+        IERC7579Module(module).onInstall(initData);
+        emit ModuleInstalled(moduleTypeId, module);
+    }
+
+    function _uninstallModule(uint256 moduleTypeId, address module, bytes memory deInitData) internal virtual {
+        if (moduleTypeId == MODULE_TYPE_VALIDATOR) {
+            require(_validators.remove(module), ERC7579Utils.ERC7579UninstalledModule(moduleTypeId, module));
+        } else if (moduleTypeId == MODULE_TYPE_EXECUTOR) {
+            require(_executors.remove(module), ERC7579Utils.ERC7579UninstalledModule(moduleTypeId, module));
+        } else if (moduleTypeId == MODULE_TYPE_FALLBACK) {
+            bytes4 selector;
+            (selector, deInitData) = abi.decode(deInitData, (bytes4, bytes));
+            require(_uninstallFallback(module, selector), ERC7579Utils.ERC7579UninstalledModule(moduleTypeId, module));
+        }
+
+        IERC7579Module(module).onUninstall(deInitData);
+        emit ModuleUninstalled(moduleTypeId, module);
+    }
+
+    function _installFallback(address module, bytes4 selector) internal virtual returns (bool) {
+        if (_fallbackHandler(selector) == address(0)) {
+            _fallbacks[selector] = module;
+            return true;
+        } else return false;
+    }
+
+    function _uninstallFallback(address module, bytes4 selector) internal virtual returns (bool) {
+        if (_fallbackHandler(selector) == module && module != address(0)) {
+            delete _fallbacks[selector];
+            return true;
+        } else return false;
+    }
+
+    function _fallbackHandler(bytes4 selector) internal view virtual returns (address) {
+        return _fallbacks[selector];
+    }
+
+    function _fallback() internal virtual {
+        address handler = _fallbackHandler(msg.sig);
+        if (handler == address(0)) revert ERC7579MissingFallbackHandler(msg.sig);
```

Comment:
Why wouldn't just `return` here? I don't find the specification to be clear about this but I'd say it's better to treat it as a noop and return right away.